juice/src/net/network.rs

```diff
@@ -39,6 +67,44 @@ impl<B: IBackend + LayerOps<f32> + 'static> Network<B> {
         Ok(Network { config, top })
     }
 
+    pub fn from_config_and_weights(
+        backend: &B,
+        into_config: impl Into<LayerConfig>,
+        input_shapes: &[TensorDesc],
+        weights: &WeightsData,
+    ) -> Result<Network<B>, NetworkFromConfigError> {
+        let net = Network::from_config(backend, into_config, input_shapes)?;
+
+        // Set weights in the network from the provided data.
+        let native_backend = native_backend();
+        let mut unused_weights: HashSet<String> = weights.weights_data.keys().cloned().collect();
+        for i in 0..net.top.descriptor().params().len() {
+            let mut to_params = net.top.descriptor().params()[i].borrow_mut();
+            match weights.weights_data.get(&to_params.path) {
+                Some(d) => {
+                    let params_data = to_params.data.write_only(native_backend.device()).unwrap();
```

Comment:
The function is already failable, so we could avoid the `unwrap`

Reply:
Done.